pkg/controller/logstash/driver.go

```diff
@@ -187,3 +193,21 @@ func podReconciliationDone(ctx context.Context, c k8s.Client, sset appsv1.Statef
 func isPendingReconciliation(sset appsv1.StatefulSet) bool {
 	return sset.Generation != sset.Status.ObservedGeneration
 }
+
+func ensureStsNameLabelIsSetOnPods(params Params) error {
+	sts, err := retrieveActualStatefulSet(params.Client, params.Logstash)
+	if apierrors.IsNotFound(err) {
+		// maybe the sts doesn't exist yet or was deleted, let it be (re)created
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	if _, ok := sts.Spec.Template.Labels[labels.StatefulSetNameLabelName]; ok {
+		// label is already in place, great
+		return nil
+	}
+	// add the missing label and update the sts resource
+	sts.Spec.Template.Labels[labels.StatefulSetNameLabelName] = params.Logstash.Name
+	return params.Client.Update(params.Context, &sts)
```

Comment:
I was wondering if we should requeue due to cache lag but I guess if would do that automatically as long as the label is missing in the cache.